templates/web/base/report/_main.html

```diff
@@ -8,19 +8,25 @@
 <div class="problem-header clearfix" problem-id="[% problem.id %]">
 
 [% IF c.user.has_permission_to('planned_reports', problem.bodies_str) %]
-<form method="post" action="/my/planned/change" id="planned_form">
+<form method="post" action="/my/planned/change" id="planned_form" class="hidden-label-target">
     <input type="hidden" name="id" value="[% problem.id %]">
     <input type="hidden" name="token" value="[% csrf_token %]">
     <input type="hidden" name="change" value="[% IF c.user.is_planned_report(problem) %]remove[% ELSE %]add[% END %]">
-    <p><input type="submit"
-        data-remove="[% loc('Remove from planned reports') %]" data-add="[% loc('Add to planned reports') %]"
-        value="
+    <p><input
+        type="submit"
+        id="shortlist-report"
+        data-label-remove="[% loc('Remove from shortlist') %]"
+        data-label-add="[% loc('Add to shortlist') %]"
+        data-value-remove="[% loc('Shortlisted') %]"
+        data-value-add="[% loc('Shortlist') %]"
     [%~ IF c.user.is_planned_report(problem) ~%]
-        [% loc('Remove from planned reports') %]
+        value="[% loc('Shortlisted') %]"
```

Comment:
I guess the text could change on the pushed-in buttons – so pushing the "Shortlist" button turns it into a pushed-in, yellow "Shortlisted"; and pushing the "Inspect" button turns it into a pushed-in, yellow "Cancel" button.

Danger here is that, while "Shortlisted" makes sense as a label (since it's describing the current state of the stateful button – it's just a linguistic equivalent to the yellow, pushed-in styling), "Cancel" is an action rather than a state, so it doesn't match. The matching label for that button, I guess, would be "Inspecting", since that describes the state the report is currently in.

Bah.